src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java

```diff
@@ -20,12 +24,19 @@
 public abstract class AbstractWebDriverTest {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AbstractWebDriverTest.class);
+    private static final String CAPABILITIES_FILE_PROP = "capabilities.config";
+    private static final String DEFAULT_CAPABILITIES_FILE = "capabilities.yml";
 
     @Parameterized.Parameters
-    public static Iterable<Object[]> data() {
-        WebDriverFactory webDriverFactory = WebDriverFactory.getInstance();
+    public static Iterable<Object[]> data() throws IOException {
+        String capabilitiesConfigFile = System.getProperty(CAPABILITIES_FILE_PROP, DEFAULT_CAPABILITIES_FILE);
+        LOGGER.debug("Using capabilities configuration from FILE :: {}", capabilitiesConfigFile);
+        URL resourceURL = AbstractWebDriverTest.class.getClassLoader().getResource(capabilitiesConfigFile);
+
+        ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+        webDriverConfiguration = objectMapper.readValue(resourceURL, WebDriverConfiguration.class);
```

Comment:
Why do we need to parse the capability file here again?

Reply:
I have removed that parsing logic from there